velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -296,6 +298,157 @@ TpchPlan TpchQueryBuilder::getQ6Plan() const {
+  core::PlanNodeId ordersScanNodeId;
+
+  auto nation =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kNation, nationSelectedRowType, nationFileColumns)
+          .capturePlanNodeId(nationScanNodeId)
+          .planNode();
+
+  auto orders = PlanBuilder(planNodeIdGenerator)
+                    .tableScan(
+                        kOrders,
+                        ordersSelectedRowType,
+                        ordersFileColumns,
+                        {orderDateFilter})
+                    .capturePlanNodeId(ordersScanNodeId)
+                    .planNode();
+
+  auto partialPlan =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kCustomer, customerSelectedRowType, customerFileColumns)
+          .capturePlanNodeId(customerScanNodeId)
+          .hashJoin(
+              {"c_custkey"},
+              {"o_custkey"},
+              orders,
+              "",
+              {"c_name",
+               "c_acctbal",
+               "c_phone",
+               "c_custkey",
+               "c_address",
+               "c_comment",
+               "c_nationkey",
+               "o_orderkey"})
+          .hashJoin(
+              {"c_nationkey"},
+              {"n_nationkey"},
+              nation,
+              "",
+              {"c_name",
+               "c_acctbal",
+               "c_phone",
+               "c_address",
+               "c_custkey",
+               "c_comment",
+               "n_name",
+               "o_orderkey"})
+          .planNode();
+
+  auto plan = PlanBuilder(planNodeIdGenerator)
+                  .tableScan(
+                      kLineitem,
+                      lineitemSelectedRowType,
+                      lineitemFileColumns,
+                      {lineitemReturnFlagFilter})
+                  .capturePlanNodeId(lineitemScanNodeId)
+                  .project(
+                      {"l_extendedprice * (1.0 - l_discount) AS part_revenue",
+                       "l_orderkey"})
+                  .hashJoin(
+                      {"l_orderkey"},
+                      {"o_orderkey"},
+                      partialPlan,
+                      "",
+                      {"part_revenue",
+                       "c_name",
+                       "c_custkey",
+                       "c_acctbal",
+                       "n_name",
+                       "c_address",
+                       "c_phone",
+                       "c_comment",
+                       "c_address"})
+                  .partialAggregation(
+                      {"c_custkey",
+                       "c_name",
+                       "c_acctbal",
+                       "n_name",
+                       "c_address",
+                       "c_phone",
+                       "c_comment"},
+                      {"sum(part_revenue) as revenue"})
+                  .localPartition({})
+                  .finalAggregation()
+                  .orderBy({"revenue DESC"}, false)
+                  .project(
```

Comment:
Yes, because revenue which is being selected is out of order with the rest of the group by keys, removing this project will result in a typecast error